src/Particles/babylon.particleSystem.ts

```diff
@@ -276,9 +341,16 @@
                 if (this._stockParticles.length !== 0) {
                     particle = this._stockParticles.pop();
                     particle.age = 0;
+                    particle.cellIndex = this.startSpriteCellID;
                 } else {
-                    particle = new Particle();
+                    if (this.spriteCellSize) {

```

Comment:
this could be avoided then as all info will be read from the particle system so no need for a custom constructor